### ISSUE TITLE
Assertion failure in jas_matrix_t (Unfixed #53 #61)

### ISSUE THREAD

**Issue by asarubbo** [mdadams/jasper#82]
On 1.900.22:

```
warning: trailing garbage in marker segment (9 bytes)
warning: trailing garbage in marker segment (40 bytes)
warning: ignoring unknown marker segment (0xffee)
type = 0xffee (UNKNOWN); len = 23;1f 32 ff ff ff 00 10 00 3d 4d 00 01 32 ff 00 e4 00 10 00 00 4f warning: trailing garbage in marker segment (34 bytes)
imginfo: /tmp/portage/media-libs/jasper-1.900.22/work/jasper-1.900.22/src/libjasper/base/jas_seq.c:90: jas_matrix_t *jas_seq2d_create(int, int, int, int): Assertion `xstart <= xend && ystart <= yend' failed.
```

Testcase: https://github.com/asarubbo/poc/blob/master/00043-jasper-assert-jas_matrix_t

**Comment by mdadams** [comment]
The root cause of this issue is not really the same bug as #53/#61.  In any case, this problem is fixed in 1.900.25.  The result in 1.900.25 is a large memory allocation that fails.  To avoid the attempt at a large allocation, I made one further change described below, which is NOT picked up in 1.900.25.

After 1.900.25 (in commit 5185cb13da413f826dcc5f1e0c1eb3e038dc0874), I added a new feature that I think I may use to replace the --memory-limit option later.  For the JP2. JPC, BMP, and JPEG decoders, the decoder will now refuse to decode an image with more than a certain number of samples (where this threshold can be set via the max_samples decoder option).  This can better handle situations where JasPer hands off to another library that does not use jas_malloc and friends and therefore does not get tracked for memory usage (which makes --memory-limit unhelpful).  I plan to add a similar functionality for the remaining decoders PNM and PGX later, since it seems like a useful feature.
